### ISSUE TITLE
Bcachefs support

### ISSUE THREAD

**Issue by Littleweisheit** [calamares/calamares#2262]
bcachefs will release with  Linux 6.7
Bcachefs has subvolume creation like btrfs and built-in encrypt.

**Comment by demmm** [comment]
It might be better to check with KPMCore, if it already supports (or plans to support) Bcachefs, since Calamares just uses what KPMCore supports.

**Comment by ptr1337** [comment]
There is already a partial commit from @vnepogodin for kpmcore:
https://invent.kde.org/vnepogodin/kpmcore/-/commit/d5aa1fe1e81a4398b7e4a7b34e0f681850a84062

**Comment by tsimonq2** [comment]
I would really like to ship this with the upcoming release.

I'm not as well-versed in KDE upstream as I should be; would it be acceptable for someone to simply cherry-pick that commit onto their own branch and finish it up?

If there's any interest, positively react to this message, I'll create a PPA for testing.

**Comment by adriaandegroot** [comment]
@stikonas ^^

**Comment by stikonas** [comment]
Kpmcore doesn't support this yet but it's on my radar. But I first need to upgrade to util-linux 2.39...

**Comment by adriaandegroot** [comment]
Closing as "needs support elsewhere".

**Comment by ptr1337** [comment]
> Closing as "needs support elsewhere".

Why not keeping that open?
Even if the kpmcore work is done, there still needs to be work done on calamares side.

**Comment by LethalManBoob** [comment]
bcache when o3o

**Comment by ptr1337** [comment]
@adriaandegroot Could you please reopen this, since kpmcore work is now mostly done?

@OpenSourceAnarchist Yes, I saw these commits, looking forward to it!

**Comment by demmm** [comment]
Since KaOS already patched kpmcore to include bcachefs, a start has been made to see what more is needed for Calamares.
It already shows as an option, but not tested yest what else needs changing:
https://github.com/KaOSx/calamares/commit/bac9da015020a4ef269401e353c921e38b77349d

**Comment by ptr1337** [comment]
@demmm 

tested it on CachyOS with Erase Disk Option, install was successfully and also booted correctly.
fstab seems correct

![image](https://github.com/calamares/calamares/assets/70081076/bd067a7a-a71f-4817-8a1a-7fe849f41ef1)

**Comment by ptr1337** [comment]
```
2024-04-06 - 16:14:47 [6]: [PYTHON JOB]: "Bootloader: grub (efi)" 
2024-04-06 - 16:14:47 [6]:     .. Running QList("grub-install", "--target=x86_64-efi", "--efi-directory=/boot/efi", "--bootloader-id=cachyos", "--force") 
2024-04-06 - 16:14:47 [6]:     .. Target cmd: QList("grub-install", "--target=x86_64-efi", "--efi-directory=/boot/efi", "--bootloader-id=cachyos", "--force") Exit code: 1 output:
 Installing for x86_64-efi platform.
grub-install: error: unknown filesystem.
2024-04-06 - 16:14:47 [2]: WARNING: [PYTHON JOB]: "Command 'grub-install --target=x86_64-efi --efi-directory=/boot/efi --bootloader-id=cachyos --force' returned non-zero exit status 1." 
2024-04-06 - 16:14:47 [6]: [PYTHON JOB]: "stdout:Installing for x86_64-efi platform.\ngrub-install: error: unknown filesystem." 
2024-04-06 - 16:14:47 [6]: [PYTHON JOB]: "stderr:None" 
2024-04-06 - 16:14:47 [6]: virtual void Calamares::JobThread::run()
```

As always, grub doesnt have support for it.

**Comment by Littleweisheit** [comment]
> ```
> 2024-04-06 - 16:14:47 [6]: [PYTHON JOB]: "Bootloader: grub (efi)" 
> 2024-04-06 - 16:14:47 [6]:     .. Running QList("grub-install", "--target=x86_64-efi", "--efi-directory=/boot/efi", "--bootloader-id=cachyos", "--force") 
> 2024-04-06 - 16:14:47 [6]:     .. Target cmd: QList("grub-install", "--target=x86_64-efi", "--efi-directory=/boot/efi", "--bootloader-id=cachyos", "--force") Exit code: 1 output:
>  Installing for x86_64-efi platform.
> grub-install: error: unknown filesystem.
> 2024-04-06 - 16:14:47 [2]: WARNING: [PYTHON JOB]: "Command 'grub-install --target=x86_64-efi --efi-directory=/boot/efi --bootloader-id=cachyos --force' returned non-zero exit status 1." 
> 2024-04-06 - 16:14:47 [6]: [PYTHON JOB]: "stdout:Installing for x86_64-efi platform.\ngrub-install: error: unknown filesystem." 
> 2024-04-06 - 16:14:47 [6]: [PYTHON JOB]: "stderr:None" 
> 2024-04-06 - 16:14:47 [6]: virtual void Calamares::JobThread::run()
> ```
> 
> As always, grub doesnt have support for it.

For now,you need use other FS for /boot

**Comment by ptr1337** [comment]
> > ```
> > 2024-04-06 - 16:14:47 [6]: [PYTHON JOB]: "Bootloader: grub (efi)" 
> > 2024-04-06 - 16:14:47 [6]:     .. Running QList("grub-install", "--target=x86_64-efi", "--efi-directory=/boot/efi", "--bootloader-id=cachyos", "--force") 
> > 2024-04-06 - 16:14:47 [6]:     .. Target cmd: QList("grub-install", "--target=x86_64-efi", "--efi-directory=/boot/efi", "--bootloader-id=cachyos", "--force") Exit code: 1 output:
> >  Installing for x86_64-efi platform.
> > grub-install: error: unknown filesystem.
> > 2024-04-06 - 16:14:47 [2]: WARNING: [PYTHON JOB]: "Command 'grub-install --target=x86_64-efi --efi-directory=/boot/efi --bootloader-id=cachyos --force' returned non-zero exit status 1." 
> > 2024-04-06 - 16:14:47 [6]: [PYTHON JOB]: "stdout:Installing for x86_64-efi platform.\ngrub-install: error: unknown filesystem." 
> > 2024-04-06 - 16:14:47 [6]: [PYTHON JOB]: "stderr:None" 
> > 2024-04-06 - 16:14:47 [6]: virtual void Calamares::JobThread::run()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > As always, grub doesnt have support for it.
> 
> For now,you need use other FS for /boot

Yes, I know. Ive just tested all bootloaders supported by Cachy. We wont switch to /boot for grub because of bcachefs.
